bodo/io/iceberg/read_compilation.py

```diff
@@ -140,33 +97,26 @@ def determine_str_as_dict_columns(
     if len(str_col_names_to_check) == 0:
         return set()  # No string as dict columns
 
-    # Get list of files. No filters are know at this time, so
+    # Get a small list of files. No filters are know at this time, so
     # no file pruning can be done.
-    # XXX We should push down some type of file limit to the
-    # Iceberg Java Library to avoid retrieving millions of files
-    # for no reason.
-    all_pq_infos = get_iceberg_file_list_parallel(
-        conn, database_schema, table_name, filters=None
-    )[0]
+    # XXX We should push down some type of file limit to the scanner
+    #     to avoid retrieving millions of files for no reason
+    all_file_tasks = list(
+        table.scan(selected_fields=tuple(str_col_names_to_check)).plan_files()
```

Comment:
Note, should be easier to do because we can access private functions from the Scanner now (cause its Python)